### ISSUE TITLE
Add monitor parameter for early stopping

### ISSUE THREAD

**Issue by candalfigomoro** [Nixtla/neuralforecast#673]
### Description

In pytorch lightning you can monitor different metrics for early stopping by setting the `monitor` parameter (e.g. https://lightning.ai/docs/pytorch/stable/common/early_stopping.html).

At the moment neuralforecast monitors the validation loss: https://github.com/Nixtla/neuralforecast/blob/3a86f1e55a7ec4a0c8ea2861ec73bd15ba5fe966/neuralforecast/common/_base_windows.py#L136

Could you add a parameter so that we can choose to monitor the validation loss or the loss (training loss)?

### Use case

The best practice is to monitor the validation loss, but in some cases it may be useful to monitor the training loss instead and in most frameworks (e.g. lightning, keras) you can choose what to monitor.

**Comment by candalfigomoro** [comment]
This is a workaround:

```
from pytorch_lightning.callbacks import EarlyStopping

# Remove the existing EarlyStopping callback and
# adds a new one that monitors the training loss
# instead of the validation loss
def change_early_stopping_monitor(model):
    # Filter out the old EarlyStopping callback
    model.trainer_kwargs["callbacks"] = [
        cb for cb in model.trainer_kwargs.get("callbacks", [])
        if not isinstance(cb, EarlyStopping)
    ]
    # Add new EarlyStopping callback
    model.trainer_kwargs["callbacks"].append(
        EarlyStopping(
            monitor="train_loss_epoch", patience=model.early_stop_patience_steps
        )
    )

nf = NeuralForecast(models=models)
change_early_stopping_monitor(nf.models[0])
```
